cmd/plugin/sync.go

```diff
@@ -0,0 +1,75 @@
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/odpf/optimus/config"
+	"github.com/odpf/optimus/plugin"
+)
+
+// NewInstallCommand initializes plugin install command
+func NewSyncCommand() *cobra.Command {
+	sync := &syncCommand{}
+	cmd := &cobra.Command{
+		Use:     "sync",
+		Short:   "sync plugins from server",
+		Example: "optimus plugin sync -c optimus.yaml",
+		PreRunE: sync.PreRunE,
+		RunE:    sync.RunE,
+	}
+	cmd.PersistentFlags().StringVarP(&sync.configFilePath, "config", "c", sync.configFilePath, "File path for optimus configuration")
+	return cmd
+}
+
+type syncCommand struct {
+	configFilePath string
+	clientConfig   *config.ClientConfig
+}
+
+func (s *syncCommand) PreRunE(_ *cobra.Command, _ []string) error {
+	c, err := config.LoadClientConfig(s.configFilePath)
+	if err != nil {
+		return err
+	}
+	s.clientConfig = c
+	return nil
+}
+
+func (s *syncCommand) downloadArchiveFromServer() error {
+	optimusServerURL := fmt.Sprintf("http://%s/%s", s.clientConfig.Host, "plugins") // will url work ?
```

Comment:
@sbchaos can we rely on this url format ?

Reply:
Can the host include the scheme ?  if we need to support https or ftp path, this will be restrictive.
Just append path `plugins` to the Host if we can accept it along with scheme.